### PR TITLE
add annotation option and a linear color map for heatmap viz.

### PR DESCRIPTION
**Pull request by xiaoyugit**
allow user to put digits on a grid heat map, which is related to #3570 and visualization for churn rate.
  
control
heatmap

### CODE REVIEW

File: superset/assets/javascripts/explore/stores/controls.jsx [hunk 1]
```diff
@@ -1086,6 +1087,14 @@ export const controls = {
     description: t('Whether to display the legend (toggles)'),
   },
 
+  show_annotation: {
```
Comment (mistercrunch) [inline superset/assets/javascripts/explore/stores/controls.jsx]: The term annotation is already used for other purposes, can we just call this `show_values`?

File: superset/assets/javascripts/explore/stores/controls.jsx [hunk 2]
```diff
@@ -1086,6 +1087,14 @@ export const controls = {
...
+    type: 'CheckboxControl',
+    label: t('Annotation'),
```
Comment (mistercrunch) [inline superset/assets/javascripts/explore/stores/controls.jsx]: `Show Values`

File: superset/assets/javascripts/explore/stores/controls.jsx [hunk 3]
```diff
@@ -1086,6 +1087,14 @@ export const controls = {
...
+    renderTrigger: true,
+    default: false,
+    description: t('Whether to display the annotation (toggles)'),
```
Comment (mistercrunch) [inline superset/assets/javascripts/explore/stores/controls.jsx]: 'Whether to display the numerical values within the cells'

File: superset/assets/visualizations/heatmap.js [hunk 4]
```diff
@@ -130,6 +130,25 @@ function heatmapVis(slice, payload) {
...
+  if (fd.show_annotation) {
+    const cells = svg.selectAll('rect')
+      .data(data)
+      .enter()
+      .append('g')
+      .attr('transform', `translate(${margin.left}, ${margin.top})`);
+
+    cells.append('text')
+      .attr('transform',
+        function (d) { return 'translate(' + xRbScale(d.x) + ',' + yRbScale(d.y) + ')'; })
```
Comment (mistercrunch) [inline superset/assets/visualizations/heatmap.js]:
```
.attr('transform', d => `translate(${xRbScale(d.x}, ${yRbScale(d.y)})`
```

File: superset/assets/visualizations/heatmap.js [hunk 5]
```diff
@@ -130,6 +130,25 @@ function heatmapVis(slice, payload) {
...
+      .attr('y', yRbScale.rangeBand() / 2)
+      .attr('x', xRbScale.rangeBand() / 2)
+      .attr('text-anchor', 'middle')
+      .attr('dy', '.35em')
+      .text(function (d) { return valueFormatter(d.v); })
```
Comment (mistercrunch) [inline superset/assets/visualizations/heatmap.js]:
```
.text(d => valueFormatter(d.v))
```

File: superset/assets/visualizations/heatmap.js [hunk 6]
```diff
@@ -130,6 +130,25 @@ function heatmapVis(slice, payload) {
...
+      .attr('font-size', Math.min(yRbScale.rangeBand(), xRbScale.rangeBand()) / 3 + 'px')
+      .attr('fill', function (d) { return d.v >= payload.data.extents[1] / 2 ? 'white' : 'black'; });
```
Comment (mistercrunch) [inline superset/assets/visualizations/heatmap.js]: arrow function here too

**Review by mistercrunch — CHANGES_REQUESTED** [review]
A few minor things, but otherwise looks good

File: superset/assets/visualizations/heatmap.js (hunk 4 above)
Comment (xiaoyugit) [reply superset/assets/visualizations/heatmap.js]: thx! using => indeed makes it neater.

File: superset/assets/javascripts/explore/stores/controls.jsx (hunk 1 above)
Comment (xiaoyugit) [reply superset/assets/javascripts/explore/stores/controls.jsx]: noted

**Review by mistercrunch — APPROVED** [review]

**Pull request merged**